scripts/tests/auth/test-private-repo-with-https-config-file.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+set -eu # don't use -x as it will leak the private key
+# shellcheck source=./setup.sh
+source "$(dirname "$0")/setup.sh"
+
+# setup earthly config
+cat << EOF > /tmp/earthconfig.https
+git:
+  github.com:
+    auth: https
+    user: cinnamonthecat
+    password: "$GITHUB_PASSWORD"
+EOF
```

Comment:
Why not use `earthly config` here?

Reply:
This test is to cover the case where a user creates the config on their own.

There's a different test under `test-private-repo-with-https-config-command.sh`  that's more or less the same as this one, but uses `earthly config` instead.